astropy/modeling/functional_models.py

```diff
@@ -311,43 +313,37 @@ class Gaussian2D(Fittable2DModel):
     y_mean = Parameter(default=0)
     x_stddev = Parameter(default=1)
     y_stddev = Parameter(default=1)
-    theta = Parameter(default=0)
+    theta = Parameter(default=0.0)
 
     def __init__(self, amplitude=amplitude.default, x_mean=x_mean.default,
                  y_mean=y_mean.default, x_stddev=None, y_stddev=None,
-                 theta=0.0, cov_matrix=None, **kwargs):
-        if y_stddev is None and cov_matrix is None:
-            raise InputParameterError(
-                "Either x/y_stddev must be specified, or a "
-                "covariance matrix.")
-        elif x_stddev is None and cov_matrix is None:
-            raise InputParameterError(
-                "Either x/y_stddev must be specified, or a "
-                "covariance matrix.")
-        elif cov_matrix is not None and (x_stddev is not None or
-                                         y_stddev is not None):
-            raise InputParameterError(
-                "Cannot specify both cov_matrix and x/y_stddev")
-
-        # Compute principle coordinate system transformation
-        elif cov_matrix is not None:
-            if (x_stddev is not None or y_stddev is not None):
-                raise InputParameterError(
-                    "Cannot specify both cov_matrix and x/y_stddev")
-            # Compute principle coordinate system transformation
-            cov_matrix = np.array(cov_matrix)
-
-            if cov_matrix.shape != (2, 2):
-                # TODO: Maybe it should be possible for the covariance matrix
-                # to be some (x, y, ..., z, 2, 2) array to be broadcast with
-                # other parameters of shape (x, y, ..., z)
-                # But that's maybe a special case to work out if/when needed
-                raise ValueError("Covariance matrix must be 2x2")
-
-            eig_vals, eig_vecs = np.linalg.eig(cov_matrix)
-            x_stddev, y_stddev = np.sqrt(eig_vals)
-            y_vec = eig_vecs[:, 0]
-            theta = np.arctan2(y_vec[1], y_vec[0])
+                 theta=None, cov_matrix=None, **kwargs):
+        if cov_matrix is None:
+            if x_stddev is None:
+                x_stddev = self.__class__.x_stddev.default
+            if y_stddev is None:
+                y_stddev = self.__class__.y_stddev.default
+            if theta is None:
+                theta = self.__class__.theta.default
+        else:
+            if x_stddev is not None or y_stddev is not None or theta is not None:
```

Comment:
I've recently seen an approach that used `list.count` for such repetitive approaches: `[x_stddev, y_stddev, theta].count(None) > 0` or if you want short-circuit behaviour then there is also `any(p is not None for p in (x_stddev, y_stddev, theta))`

That's a totally stylistic point and not a request but I realized you exceeded the 80 character line limit.